sv.c

```diff
@@ -4616,9 +4632,9 @@ Perl_sv_setsv_flags(pTHX_ SV *dsv, SV* ssv, const I32 flags)
          *
          */
 
-        /* Whichever path we take through the next code, we want this true,
-           and doing it now facilitates the COW check.  */
-        (void)SvPOK_only(dsv);
+        (void)SvOK_off(dsv);
+        SvFLAGS(dsv) |= sflags &
+            (SVf_POK|SVp_POK|SVf_IOK|SVp_IOK|SVf_NOK|SVp_NOK|SVf_UTF8);
```

Comment:
why isnt `SVF_ISUV` copied over from `sflags`? did anything earlier wipe the `SVf_UTF8` inside `SvFLAGS(dsv)`

Reply:
Thanks for pointing this out, I think the flag handling around this bit of code could be tidied up further. Will look at it again.

Reply:
Remember `(void)SvOK_off(dsv);` being a func call was IIRC a Father C 2014 or 2012 quick fix for random memory corruption/leaks involving `SvOOK()` back when the OOK delta count was stored in a `SVt_PVIV` inside field `SvIVX()`. Someone later on changed `OOK()`'s design to be a BER/ASN.1/SMI/compressed integer stored in the PV buffer. Eliminating the former design limitation, that an OOK POK SV, can NEVER EVER EVER be a dual var with POK + OOK + IOK.

I'm not gonna touch cactus that is `sv_backoff()`/`SvOK_off(dsv)` unless I'm at a YAPC or a Toolchain summit with another qualified Perl C human 3-6 feet away.

Reply:
`SVf_IVisUV` will be copied over in the imminent push.